test/unit/test-reader/set-collection.test.js

```diff
@@ -0,0 +1,122 @@
+'use strict';
+
+const _ = require('lodash');
+const q = require('q');
+const pathUtils = require('lib/test-reader/path-utils');
+const SetCollection = require('lib/test-reader/set-collection');
+const Set = require('lib/test-reader/set');
+
+describe('set-collection', () => {
+    const sandbox = sinon.sandbox.create();
+
+    let set;
+
+    const stubConfig = (opts) => {
+        opts = _.defaults(opts || {}, {
+            set1: {},
+            set2: {}
+        });
+
+        return {
+            sets: {
+                set1: opts.set1,
+                set2: opts.set2
+            },
+            system: {
+                projectRoot: '/root'
+            }
+        };
+    };
+
+    beforeEach(() => {
+        set = sandbox.stub();
+        set.filterFiles = sandbox.stub();
+
+        sandbox.stub(pathUtils, 'expandPaths').returns(q([]));
+    });
+
+    afterEach(() => sandbox.restore());
+
+    it('should be created with sets specified from the cli option', () => {
```

Comment:
`should create collection for specified sets`
`should create collection for config sets by default`